mergify_engine/worker.py

```diff
@@ -142,11 +142,35 @@ class UnexpectedPullRetry(Exception):
 T_MessageID = typing.NewType("T_MessageID", str)
 
 
-def get_low_priority_minimal_score() -> float:
-    # NOTE(sileht): score is scheduled_at timestamp for high
-    # prio and scheduled_at timestamp * 10 for low prio, pick *
-    # 2 to split the bucket
-    return date.utcnow().timestamp() * 2
+# NOTE(sileht): score is:
+# * scheduled_at timestamp for high prio
+# * scheduled_at timestamp * 10 for medium prio
+# * scheduled_at timestamp * 100 for low prio
+
+
+def get_high_priority_score() -> float:
+    return date.utcnow().timestamp()
+
+
+def get_medium_priority_score() -> float:
+    return date.utcnow().timestamp() * 10
+
+
+def get_low_priority_score() -> float:
+    return date.utcnow().timestamp() * 100
+
+
+def get_priority_level_from_score(
+    score: float,
+) -> typing.Literal["high", "medium", "low"]:
+    low_prio_threshold = get_low_priority_score() * 2
```

Comment:
So if the score is has just been set to `date.utcnow().timestamp() * 100` (low priority score), when going through this function it will have a priority level of "medium". I might be missing something but i don't see any cases where an event could have a level of "low" in this, is this what is wanted ?

Reply:
I'm a bit concerned also by the lack of unit tests here. 😄

Reply:
What is stored in the ordered set is not the list of events, but just the PR to refresh. The events list is stored in another Redis key (one by PR). 
Here is a beautiful ASCII art of the data structures: https://github.com/Mergifyio/mergify-engine/blob/main/mergify_engine/worker.py#L15-L40 :D

So if a PR is already in the set with a higher score (a lower prior), and a new event comes for the same PR but with a lower score (higher prior), we add this event in the other key and just override the score (the position) in the set.

If we use 3 lists/sets for this, if the priority to refreshing a PR change, we need to clean up the two other lists/sets.

But even if there is a little overhead as maintaining 3 lists/sets instead of 1, maybe this is worth it for readability... What does other think?

I'm just a bit fearful to change how the priority system currently works (Lua scripts and the polling system), which is a bigger task.

Here, I just wanted to introduce a new level of priority.

Reply:
Using an offset+divider already makes priorities clearer to handle and read. Like @sileht said, further changes would lead to a bigger rework than what we actually need for this PR to fit, even though leveraging sets is a good idea for handling priorities!

Reply:
Yeah but in 5138 Mergify would go boom with this solution :(